clarifai/models/model_serving/model_config/base.py

```diff
@@ -0,0 +1,323 @@
+from typing import Dict, Iterable, List, TypedDict, Union
+
+import numpy as np
+
+from .config import ModelConfigClass, ModelTypes, get_model_config
+from .output import (ClassifierOutput, EmbeddingOutput, ImageOutput, MasksOutput, TextOutput,
+                     VisualDetectorOutput)
+from .triton import wrappers as triton_wrapper
+
+
+class _TypeCheckModelOutput(type):
+
+  def __new__(cls, name, bases, attrs):
+    """
+    Override child `predict` function with parent._output_type_check(child.predict).
+    Aim to check if child.predict returns valid output type
+    """
+
+    def wrap_function(fn_name, base, base_fn, other_fn):
```

Comment:
Thanks for your suggestions. The goal is to allow user to run the implementation just like normal python script (outside triton env - model.py) so all triton configs are kept out of working directory and no need to worry about backend-triton. Here I try to enforce user focus on instruction in docstring by checking input args and output type.
I also think user should return standard types but since we need the output has specific dims and value so I think better to keep using current output types.

I will change [this function](https://github.com/Clarifai/clarifai-python/blob/5472092c4b907c6457bbbb621b9aab2be9a12b09/clarifai/models/model_serving/model_config/base.py#L114) to vanilla call like PR #277

Reply:
Thanks --- that all sounds good.  The InferenceOutput classes seem OK to me, as they are basically just named tuples of arrays and strings, so simple to understand.